dvc/output/local.py

```diff
@@ -16,82 +15,52 @@
 
 class OutputLOCAL(OutputBase):
     REMOTE = RemoteLOCAL
+    sep = os.sep
 
-    def __init__(
-        self,
-        stage,
-        path,
-        info=None,
-        remote=None,
-        cache=True,
-        metric=False,
-        persist=False,
-        tags=None,
-    ):
-        super(OutputLOCAL, self).__init__(
-            stage,
-            path,
-            info,
-            remote=remote,
-            cache=cache,
-            metric=metric,
-            persist=persist,
-            tags=tags,
-        )
-        if remote:
-            p = os.path.join(
-                remote.prefix, urlparse(self.url).path.lstrip("/")
-            )
+    def _parse_path(self, remote, path):
+        parsed = urlparse(path)
+        if parsed.scheme == "remote":
+            p = remote.path_info / parsed.path.lstrip("/")
         else:
-            p = path
-
-        if not os.path.isabs(p):
-            p = self.remote.to_ospath(p)
-            p = os.path.join(stage.wdir, p)
-        p = os.path.abspath(os.path.normpath(p))
+            p = self.REMOTE.path_cls.from_posix(path)
+            if not p.is_absolute():
+                p = self.stage.wdir / p
 
-        self.path_info = PathLOCAL(url=self.url, path=p)
+        abs_p = os.path.abspath(os.path.normpath(fspath_py35(p)))
```

Comment:
can we make abs abs path to be part of the Path class. It looks too complicated - we are translating from posix, do some modification, back to string to normalize.

Reply:
`pathlib` contains pure and real paths and only the latter have access to filesystem. In dvc we also have urls represented by `URLInfo` now, which are pure out of necessity, e.g. there are things like key files and other params that don't go to url. 

So we have pure url, which do parsing, component access, path/url arithmetic, comparison, representation, etc and remotes which do any real world access. I follow this separation in `PathInfo`/`RemoteLOCAL` for consistency. This is why `PathInfo` can't do `normpath()/abspath()` - first resolves symlinks and thus reads fs and second accesses curdir.